### ISSUE TITLE
Make `argInfoOverlappable` a part of `Instance`

### ISSUE THREAD

**Issue by andreasabel** [agda/agda#2399]
@UlfNorell: as the bit `argInfoOverlappable` only makes sense when `argInfoHiding == Instance`, please refine the constructor `Instance` of the `Hiding` type to `Instance :: Bool -> Hiding`.
```haskell
data ArgInfo = ArgInfo
  { argInfoHiding       :: Hiding
  , argInfoRelevance    :: Relevance
  , argInfoOrigin       :: Origin
  , argInfoOverlappable :: Bool
  } deriving (Typeable, Eq, Ord, Show)
```

**Comment by gallais** [comment]
This was fixed in 63389f8201b3c9cc12252ee15125d43e2317b9e7

**Comment by andreasabel** [comment]
Thanks for fixing and locating the fix, resp.
